
This works around some Makefile limitations I want not invest time in fixing
better ...

  - Rene Rebe <rene@exactcode.de>


diff -ur blender-2.34/extern/Makefile blender-2.34-hacked/extern/Makefile
--- blender-2.34/extern/Makefile	2004-07-17 06:59:25.000000000 +0200
+++ blender-2.34-hacked/extern/Makefile	2004-09-27 14:47:38.676018120 +0200
@@ -34,7 +34,7 @@
 
 SOURCEDIR = extern
 DIR = $(OCGDIR)/extern
-DIRS = qhull/src solid
+DIRS = qhull/src solid ode
 
 TARGET =
 ifneq ($(OS),irix)
diff -ur blender-2.34/source/Makefile blender-2.34-hacked/source/Makefile
--- blender-2.34/source/Makefile	2004-07-19 04:43:07.000000000 +0200
+++ blender-2.34-hacked/source/Makefile	2004-09-27 15:58:27.158150664 +0200
@@ -367,36 +367,12 @@
     NAN_SND_LIBS += $(SOUNDSYSTEM)
 else
   ifeq ($(OS),$(findstring $(OS), "freebsd linux windows"))
-    ifeq ($(CPU),$(findstring $(CPU), "i386 powerpc"))
-      ifeq ($(OS),freebsd)
-        NAN_SND_LIBS =  $(SOUNDSYSTEM)
-        NAN_SND_LIBS += $(DUMMYSOUND)
-        NAN_SND_LIBS += $(OPENALSOUND)
-        NAN_SND_LIBS += $(SDLSOUND)
-        NAN_SND_LIBS += $(NAN_OPENAL)/lib/libopenal.a
-        NAN_SND_LIBS += $(SOUNDSYSTEM)
-      else
-        NAN_SND_LIBS =  $(SOUNDSYSTEM)
-        NAN_SND_LIBS += $(DUMMYSOUND)
-        NAN_SND_LIBS += $(OPENALSOUND)
-        NAN_SND_LIBS += $(SDLSOUND)
-        NAN_SND_LIBS += $(NAN_OPENAL)/lib/libopenal.a
-        NAN_SND_LIBS += $(SOUNDSYSTEM)
-      endif
-    else
-      ifeq ($(OS),windows)
-        NAN_SND_LIBS =  $(SOUNDSYSTEM)
-        NAN_SND_LIBS += $(DUMMYSOUND)
-        NAN_SND_LIBS += $(OPENALSOUND)
-        NAN_SND_LIBS += $(SDLSOUND)
-        NAN_SND_LIBS += $(NAN_OPENAL)/lib/openal_static.lib
-        NAN_SND_LIBS += $(SOUNDSYSTEM)
-      else
-        NAN_SND_LIBS =  $(SOUNDSYSTEM)
-        NAN_SND_LIBS += $(DUMMYSOUND)
-        NAN_SND_LIBS += $(SOUNDSYSTEM)
-      endif
-    endif
+      NAN_SND_LIBS =  $(SOUNDSYSTEM)
+      NAN_SND_LIBS += $(DUMMYSOUND)
+      NAN_SND_LIBS += $(OPENALSOUND)
+      NAN_SND_LIBS += $(SDLSOUND)
+      NAN_SND_LIBS += -lopenal
+      NAN_SND_LIBS += $(SOUNDSYSTEM)
   else
     ifeq ($(OS),darwin)
       NAN_SND_LIBS =  $(SOUNDSYSTEM)
diff -ur blender-2.34/source/nan_link.mk blender-2.34-hacked/source/nan_link.mk
--- blender-2.34/source/nan_link.mk	2004-07-28 11:31:36.000000000 +0200
+++ blender-2.34-hacked/source/nan_link.mk	2004-09-27 16:19:38.580865032 +0200
@@ -87,29 +87,14 @@
 endif
 
 ifeq ($(OS),linux)
-  ifeq ($(CPU),alpha)
-    COMMENT = "MESA 3.1"
-    LLIBS = -lGL -lGLU -L/usr/X11R6/lib/ -lXmu -lXext -lX11
-    LLIBS += -lc -lm -ldl -lutil
-    LOPTS = -export-dynamic
-  endif
-  ifeq ($(CPU),i386)
     COMMENT = "MESA 3.1"
     LLIBS = -L$(NAN_MESA)/lib -L/usr/X11R6/lib -lXmu -lXext -lX11 -lXi
     LLIBS += -lutil -lc -lm -ldl -lpthread
     LLIBS += -L$(NAN_ODE)/lib -lode
     LOPTS = -export-dynamic
     DADD = -lGL -lGLU
-    SADD = $(NAN_MESA)/lib/libGL.a $(NAN_MESA)/lib/libGLU.a
+    SADD = -lGL -lGLU
     DYNLDFLAGS = -shared $(LDFLAGS)
-  endif
-  ifeq ($(CPU),$(findstring $(CPU), "powerpc sparc64"))
-    LLIBS = -L/usr/X11R6/lib/ -lXmu -lXext -lX11 -lc -ldl -lm -lutil
-    DADD = -lGL -lGLU
-    SADD = /usr/lib/libGL.a /usr/lib/libGLU.a
-    LOPTS = -export-dynamic
-	DYNLDFLAGS = -shared $(LDFLAGS)
-  endif
     LLIBS += -lz
 endif
 
